### PR TITLE
FOUR-8793: Script Security Logs does not provide user creator information

### PR DESCRIPTION
**Pull request by CarliPinell**
## Issue & Reproduction Steps
Expected Behavior:
QA observations
Please add the date of creation, description, category and language

## Related Tickets & Packages
- FOUR-8793

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy

### CODE REVIEW

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-ba43526259.eng.processmaker.net

File: ProcessMaker/Events/ScriptDeleted.php [hunk 1]
```diff
@@ -13,6 +14,7 @@ class ScriptDeleted implements SecurityLogEventInterface
     use FormatSecurityLogChanges;
 
     private Script $script;
+    public string $userDelete;
```
Comment (pmPaulis) [inline ProcessMaker/Events/ScriptDeleted.php:17]: Please remove string $userDelete is not used

File: ProcessMaker/Events/ScriptCreated.php [hunk 2]
```diff
@@ -13,9 +14,8 @@ class ScriptCreated implements SecurityLogEventInterface
     use FormatSecurityLogChanges;
 
     private array $changes;
-
     private array $original;
-
+    public string $categoryName;
```
Comment (pmPaulis) [inline ProcessMaker/Events/ScriptCreated.php:18]: `private string $categoryName = '';`

File: ProcessMaker/Events/ScriptCreated.php [hunk 3]
```diff
@@ -28,6 +28,8 @@ public function __construct(Script $script, array $changes)
     {
         $this->script = $script;
         $this->changes = $changes;
+        $categoryId = $this->script['script_category_id'] ?? '';
```
Comment (pmPaulis) [inline ProcessMaker/Events/ScriptCreated.php:31]: please use this:
```
$categoryId = $this->script['script_category_id'] ?? '';
if (!empty($categoryId)) {
  $this->categoryName = ScriptCategory::where('id', $categoryId)->value('name');
...
```

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Events/ScriptDeleted.php (hunk 1 above)
Comment (CarliPinell) [reply ProcessMaker/Events/ScriptDeleted.php:17]: Solved

File: ProcessMaker/Events/ScriptCreated.php (hunk 2 above)
Comment (CarliPinell) [reply ProcessMaker/Events/ScriptCreated.php:18]: Solved.

File: ProcessMaker/Events/ScriptCreated.php (hunk 3 above)
Comment (CarliPinell) [reply ProcessMaker/Events/ScriptCreated.php:31]: Solved.

**Review by pmPaulis — APPROVED** [review]

**Pull request merged**